pkg/imagesystem/registrytemplate.go

```diff
@@ -38,7 +39,7 @@ func registryService(namespace string) []client.Object {
 	}
 }
 
-func registryDeployment(namespace, registryImage string) []client.Object {
+func registryDeployment(namespace, registryImage string, cfg *apiv1.Config) []client.Object {
```

Comment:
nit: This function only really depends on a small portion of `Config`; a more concise  signature would look something like this:

```go
func registryDeployment(namespace, registryImage string, requirements corev1.ResourceRequirements) []client.Object
```

(see also https://en.wikipedia.org/wiki/Interface_segregation_principle)

Reply:
I believe this is addressed. What do you think?